packages/flutter/lib/src/rendering/sliver_decoration.dart

```diff
@@ -0,0 +1,122 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/widgets.dart';
+
+import 'object.dart';
+import 'proxy_sliver.dart';
+import 'sliver.dart';
+
+/// Paints a [Decoration] either before or after its child paints.
+class RenderSliverDecoration extends RenderProxySliver {
```

Comment:
Nice using the RenderProxySliver. Makes for a really clean and simple implementation. 💯